examples/block_writer/main.go

```diff
@@ -0,0 +1,202 @@
+		if _, err := bw.db.Exec(insert, bw.id, blockID, blockData); err != nil {
+			if log.V(1) {
+				log.Warning(err)
+			}
+			continue
+		}
+		atomic.AddUint64(&numBlocks, 1)
+	}
+}
+
+// randomBlock generates a slice of randomized bytes. Random data is preferred
+// ot prevent any sort of compression in storage.
+func (bw *blockWriter) randomBlock() []byte {
+	blockSize := bw.rand.Intn(*maxBlockSizeBytes-*minBlockSizeBytes) + *minBlockSizeBytes
+	blockData := make([]byte, blockSize)
+	for i := range blockData {
+		blockData[i] = byte(bw.rand.Int() & 0xff)
+	}
+	return blockData
+}
+
+// parseFlags ensures that values taken from command line flags are valid.
+func parseFlags() {
+	flag.Parse()
+	if *concurrency < 1 {
+		log.Fatal("Value of 'concurrency' flag must be greater than or equal to 1")
+	}
+	if *maxBlockSizeBytes < *minBlockSizeBytes {
+		log.Fatal("Value of 'max-block-bytes' must be greater than or equal to value of 'min-block-bytes'")
+	}
+}
+
+// maybeStartLocalServer will start a temporary local server if no database URL has been
+// provided.
+func maybeStartLocalServer() *server.TestServer {
+	if *dbURL != "" {
+		return nil
+	}
+	// Start a local test server.
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	*dbURL = "https://root@" + serv.ServingAddr() + "?certs=test_certs"
+	return serv
+}
+
+// setupDatabase performs initial setup for the example, creating a database and
+// with a single table. If the desired table already exists on the cluster, the
+// existing table will be dropped.
+func setupDatabase() *sql.DB {
+	var db *sql.DB
+	parsedURL, err := url.Parse(*dbURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Remove database from parsedUrl
+	q := parsedURL.Query()
+	q.Del("database")
+	parsedURL.RawQuery = q.Encode()
+
+	// Open connection to server and create a database.
+	if db, err = sql.Open("cockroach", parsedURL.String()); err != nil {
+		log.Fatal(err)
+	}
+	if _, err = db.Exec("CREATE DATABASE datablocks"); err != nil {
+		// Database might already exist, just log a warning.
+		log.Warning(err)
+	}
+	db.Close()
+
+	// Open connection directly to the new database.
+	q.Add("database", "datablocks")
+	parsedURL.RawQuery = q.Encode()
+	if db, err = sql.Open("cockroach", parsedURL.String()); err != nil {
+		log.Fatal(err)
+	}
+	// Allow a maximum of concurrency+1 connections to the database.
+	db.SetMaxOpenConns(*concurrency + 1)
+
+	// Create the initial table for storing blocks.
+	if _, err = db.Exec(`DROP TABLE IF EXISTS blocks`); err != nil {
+		log.Fatal(err)
+	}
+	if _, err = db.Exec(`
+	CREATE TABLE IF NOT EXISTS blocks (
+	  writer_id STRING NOT NULL, 
+	  block_id BIGINT NOT NULL, 
+	  raw_bytes BYTES NOT NULL, 
+	  PRIMARY KEY (writer_id, block_id)
+	)`); err != nil {
+		log.Fatal(err)
+	}
+
+	return db
+}
+
+func main() {
+	parseFlags()
+
+	serv := maybeStartLocalServer()
+	if serv != nil {
+		defer serv.Stop()
+	}
+
+	db := setupDatabase()
```

Comment:
i'd put all the `log.{Fatal,Error}` calls in `main`, with other functions returning `util.Errorf`s. I think it makes it easier to:
- reason about error cases (particularly fatal ones)
- replace `main` with `BenchmarkFoo`